src/winrm_client.py

```diff
@@ -0,0 +1,103 @@
+import base64
+import logging
+import os
+import winrm
+from src.base_client import BaseShellClient
+from scripts.powershell_scripts import (
+    create_directory_script,
+    upload_file_script,
+    file_exists_check_script,
+    get_file_checksum_script
+)
+from src.utils import calculate_checksum
+
+
+class WinRMClient(BaseShellClient):
+    def __init__(self, host, username, password):
+        super().__init__(host, username, password)
+        self.session = winrm.Session(host, auth=(username, password), transport='basic', server_cert_validation='ignore')
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        logging.info(f"Disconnected from {self.host}.")
+        pass
+
+    @staticmethod
+    def read_and_encode_file(local_path):
+        if not os.path.exists(local_path):
+            raise FileNotFoundError(f"The file {local_path} does not exist.")
+        with open(local_path, 'rb') as file:
+            content = file.read()
+        return base64.b64encode(content).decode('utf-8')
+
+    def create_remote_directory(self, directory):
+        create_dir_script = create_directory_script(directory)
+        result = self.session.run_ps(create_dir_script)
+        if result.std_err:
+            raise RuntimeError(f"Error creating directory: {result.std_err.decode().strip()}")
+
+    def check_remote_directory_exists(self, directory):
+        dir_exists_script = file_exists_check_script(directory)
+        result = self.session.run_ps(dir_exists_script)
+        if result.std_err:
+            raise RuntimeError(f"Error checking directory: {result.std_err.decode().strip()}")
+        return result.std_out.decode().strip().lower() == 'true'
+
+    def upload_file_content(self, encoded_content, remote_path):
+        upload_script = upload_file_script(encoded_content, remote_path)
+        result = self.session.run_ps(upload_script)
+        if result.std_err:
+            raise RuntimeError(f"Error uploading file: {result.std_err.decode().strip()}")
+        if not self.check_remote_directory_exists(remote_path):
+            raise FileNotFoundError(f"File {remote_path} does not exist after upload.")
+
+    def upload_file(self, local_path, remote_path):
```

Comment:
Implicit behaviour. I expect that function will upload file from local path to remote dir
```suggestion
    def upload_file(self, local_path, remote_dir):
         if not self.is_directory_exist(remote_dir):
             self.create_directory(directory)
         upload_script = upload_file_script(encoded_content, remote_path)
         result = self.session.run_ps(upload_script)
         if result.std_err:
             raise RuntimeError(f"Error uploading file: {result.std_err.decode().strip()}")


```